src/lib/marshal/translation/writer.ts

```diff
@@ -8,19 +8,40 @@ import { DirContext } from "@/lib/helpers/fs";
 import { DOUBLE_SPACES } from "@/lib/helpers/json";
 import { TranslationDirContext } from "@/lib/run-context";
 
-import { buildTranslationFileCtx, TranslationFileContext } from "./helpers";
+import {
+  buildTranslationFileCtx,
+  TranslationFileContext,
+  TranslationFormat,
+} from "./helpers";
 import { TranslationData } from "./types";
 
+type WriteTranslationFileOpts = {
```

Comment:
```suggestion
// TODO: Extend and use this options everywhere rather than re-defining opts.
type WriteTranslationFileOpts = {
```
Would you mind leaving this note above? I don't love that we are defining the same thing over and over again right now.

Reply:
Do you mean have a generic `WriteFileOpts` or do you mean a generic translation options type?

Reply:
Yeah i was thinking just using `WriteTranslationFileOpts` everywhere we take the same options (we could rename it to something more general like `TranslationFileOpts`).

And if/when at some point a function higher up needs to add other options, we can define a new type by extending from the `TranslationFileOpts` type, like:
```
type SomeNewOpts = TranslationFileOpts & {
  foo?: boolean
}
```
That way we can still just pass the same opts object down, since `SomeNewOpts` would be a superset of `TranslationFileOpts`.